packages/example/src/app/layout.tsx

```diff
@@ -0,0 +1,29 @@
+import {ReactNode} from 'react';
+// import Provider from './Provider';
+import ServerOnlyContext from './ServerOnlyContext';
+
+type Props = {
+  children: ReactNode;
+};
+
+export default function RootLayout({children, ...rest}: Props) {
+  console.log(rest);
+
+  // How to get this from the URL?
```

Comment:
Maybe wrap all pages under optional `[[locale]]` then get using `params`

https://beta.nextjs.org/docs/routing/defining-routes#example

Reply:
Hey and thank you for your interest! Yep, that seems to work, I've experimented with it here: https://github.com/amannn/next-intl/blob/ccb9c8afabc962f043c731677d2f68f871e01836/packages/example/src/app/%5Blocale%5D/page.tsx#L11-L15

I think that part is doable, the part where I currently don't see an ergonomic solution is how global configuration like messages can be passed to the library (see the topic about context in the PR description).

Reply:
I've tried some workarounds, 
1, Using singleton — Failed because for RSC, initializing on root layout requires 2 render to assign the `messages` object.
2. Using third-party state manager like zustand — Failed because of the same reason as number 1.

Probably the solution is gonna be around client component. To use in a server component we could provide utility component like `<T key='dashboard.hi' />`. Performance-wise, next will also server-render client components so the downside is on developer experience. But won't be much I think.